### PR TITLE
deprecate cpad

### PR DESCRIPTION
**Pull request by fredrikekre**
Alternative to #22184 / #22186

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

File: base/deprecated.jl [hunk 1]
```diff
@@ -1642,6 +1642,9 @@ end
     end
 end
 
+# PR #23187
+@deprecate cpad(s, n::Integer, p=" ") rpad(lpad(s,div(n+strwidth(s),2),p),n,p) false
```
Comment (Sacha0) [inline base/deprecated.jl]: A few more spaces might improve readability? :)

File: base/pkg/entry.jl [hunk 2]
```diff
@@ -572,7 +572,8 @@ end
 
 function warnbanner(msg...; label="[ WARNING ]", prefix="")
     cols = Base.displaysize(STDERR)[2]
-    warn(prefix="", Base.cpad(label,cols,"="))
+    str = rpad(lpad(label,div(cols+strwidth(label),2),"="),cols,"=")
```
Comment (Sacha0) [inline base/pkg/entry.jl]: Likewise here, a few spaces might improve readability :).

File: NEWS.md [hunk 3]
```diff
@@ -299,6 +299,8 @@ Deprecated or removed
   * Calling `union` with no arguments is deprecated; construct an empty set with an appropriate
     element type using `Set{T}()` instead ([#23144]).
 
+  * `Base.cpad` is removed; use appropriate combination of `rpad` and `lpad` instead ([#23187]).
```
Comment (Sacha0) [inline NEWS.md]: Perhaps "is removed" -> "has been removed" or "is deprecated", and perhaps "use appropriate" -> "use an appropriate"? :)

**Review by Sacha0 — APPROVED** [review]
Thanks for the followup review request! That's a great approach :).

**Comment by fredrikekre** [comment]
Double time-out on travis...

**Pull request merged**